SConstruct

```diff
@@ -531,141 +535,176 @@ class ObjectBuilder(object):
             if kwds:
                 env = env.Clone()
                 env.Prepend(**kwds)
-            path = UNITY_BUILD_DIRECTORY + filename
-            o = env.Object(Beast.variantFile(path, self.variant_dirs))
+            o = env.Object(Beast.variantFile(filename, self.variant_dirs))
             self.objects.append(o)
 
+def list_sources(base, suffixes):
+    def _iter(base):
+        for parent, dirs, files in os.walk(base):
+            files = [f for f in files if not f[0] == '.']
+            dirs[:] = [d for d in dirs if not d[0] == '.']
```

Comment:
No, it's a cool feature.  I knew about it but I'd never seen it used - at first I was wondering why you were assigning to `dirs[:]` instead of just `dirs` but I remembered that parameter.

It's sort of weird, but then it's pretty logical too.  "Here's a list of what I'm going to recurse into, remove parts you don't want."